File: solutions/r/raindrops/1/raindrops.R


raindrops <- function(number) {
    res <-   ""
    if (number %% 3 == 0) {
        res <- paste0(res,"Pling")
    }
    if (number %% 5 == 0) {
        res <- paste0(res, "Plang")
    }
    if (number %% 7 == 0) {
        res <- paste0(res,"Plong")
    }
    if (res == "") {
        res <- sprintf("%d", number)
    }
    res
}

File: solutions/ruby/raindrops/1/raindrops.rb
class Raindrops
  def Raindrops.convert(number)
    res = ""
    if (number % 3 == 0) then
      res += "Pling"
    end
    if (number % 5 == 0) then
      res += "Plang"
    end
    if (number % 7 == 0) then
      res += "Plong"
    end
    if (res == "") then
      res += "#{number}"
    end
    res
  end
end

